Intermittent error in pysph.parallel.tests.test_parallel.MPIReduceArrayTestCase.test_parallel_reduce

The test `pysph.parallel.tests.test_parallel.MPIReduceArrayTestCase.test_parallel_reduce` fails form time to time in the debian-ci infrastructure.
The error does not happen at every run so it is quite hard to track down.

Please find below the test output:

```sh
============================= test session starts ==============================
platform linux -- Python 3.9.9, pytest-6.2.5, py-1.10.0, pluggy-0.13.0
rootdir: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd, configfile: tox.ini
collected 889 items / 90 deselected / 799 selected

pysph/base/tests/test_device_helper.py .ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss [  4%]
.ss.ss.ss.ss.ss.ss.sss                                                   [  6%]
pysph/base/tests/test_domain_manager.py sss.........                     [  8%]
pysph/base/tests/test_kernel.py ........................................ [ 13%]
.................................................................        [ 21%]
pysph/base/tests/test_linalg3.py .......                                 [ 22%]
pysph/base/tests/test_neighbor_cache.py ....                             [ 22%]
pysph/base/tests/test_nnps.py .......ssssssssss......................... [ 28%]
....................................................ssssssssssssssssssss [ 37%]
ssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss........ [ 46%]
.........................................                                [ 51%]
pysph/base/tests/test_octree.py .....s...s...s...s...s...s...s...s.      [ 55%]
pysph/base/tests/test_particle_array.py ................................ [ 59%]
...ssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss        [ 67%]
pysph/base/tests/test_periodic_nnps.py ............                      [ 69%]
pysph/base/tests/test_reduce_array.py ......                             [ 70%]
pysph/base/tests/test_utils.py ..                                        [ 70%]
pysph/examples/tests/test_riemann_solver.py s........                    [ 71%]
pysph/parallel/tests/test_openmp.py .                                    [ 71%]
pysph/parallel/tests/test_parallel.py .ss.F.                             [ 72%]
pysph/parallel/tests/test_parallel_run.py .                              [ 72%]
pysph/solver/tests/test_application.py ...                               [ 72%]
pysph/solver/tests/test_solver.py ..                                     [ 73%]
pysph/solver/tests/test_solver_utils.py ............                     [ 74%]
pysph/sph/bc/tests/test_simple_inlet_outlet.py .......                   [ 75%]
pysph/sph/tests/test_acceleration_eval.py ....................ssssssssss [ 79%]
ssssssssssssssssssssssssssssssssssssssssssssss                           [ 84%]
pysph/sph/tests/test_acceleration_eval_cython_helper.py ...              [ 85%]
pysph/sph/tests/test_equations.py ...............                        [ 87%]
pysph/sph/tests/test_integrator.py ...............ssssssss               [ 90%]
pysph/sph/tests/test_integrator_cython_helper.py .                       [ 90%]
pysph/sph/tests/test_integrator_step.py ..                               [ 90%]
pysph/sph/tests/test_kernel_corrections.py ................              [ 92%]
pysph/sph/tests/test_linalg.py .............                             [ 94%]
pysph/sph/tests/test_multi_group_integrator.py .s                        [ 94%]
pysph/sph/tests/test_riemann_solver.py .............                     [ 95%]
pysph/sph/tests/test_scheme.py .                                         [ 96%]
pysph/tools/tests/test_geometry.py .............s...                     [ 98%]
pysph/tools/tests/test_geometry_stl.py ...........                       [ 99%]
pysph/tools/tests/test_interpolator.py ...                               [100%]

=================================== FAILURES ===================================
_________________ MPIReduceArrayTestCase.test_parallel_reduce __________________

self = <pysph.parallel.tests.test_parallel.MPIReduceArrayTestCase testMethod=test_parallel_reduce>

    @mark.parallel
    def test_parallel_reduce(self):
        args = ['--directory=%s' % self.root]
>       run_parallel_script.run(
            filename='simple_reduction.py', args=args, nprocs=nprocs, path=path
        )

pysph/parallel/tests/test_parallel.py:101: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

filename = 'simple_reduction.py', args = ['--directory=/tmp/tmpag842xi0']
nprocs = 2, timeout = 30.0
path = '/build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/parallel/tests'

    def run(filename, args=None, nprocs=2, timeout=30.0, path=None):
        """Run a python script with MPI or in serial (if nprocs=1).  Kill process
        if it takes longer than the specified timeout.
    
        Parameters:
        -----------
        filename - filename of python script to run under mpi.
        args - List of arguments to pass to script.
        nprocs - number of processes to run (1 => serial non-mpi run).
        timeout - time in seconds to wait for the script to finish running,
            else raise a RuntimeError exception.
        path - the path under which the script is located
            Defaults to the location of this file (__file__), not curdir.
    
        """
        if args is None:
            args = []
        file_path = abspath(join(path, filename))
        cmd = [sys.executable, file_path] + args
        if nprocs > 1:
            cmd = ['mpiexec', '-n', str(nprocs)] + cmd
    
        print('running test:', cmd)
    
        process = Popen(cmd, stdout=PIPE, stderr=PIPE)
        timer = Timer(timeout, kill_process, [process])
        timer.start()
        out, err = process.communicate()
        timer.cancel()
        retcode = process.returncode
        if retcode:
            msg = 'test ' + filename + ' failed with returncode ' + str(retcode)
            print(out.decode('utf-8'))
            print(err.decode('utf-8'))
            print('#'*80)
            print(msg)
            print('#'*80)
>           raise RuntimeError(msg)
E           RuntimeError: test simple_reduction.py failed with returncode -9

pysph/tools/run_parallel_script.py:54: RuntimeError
----------------------------- Captured stdout call -----------------------------
running test: ['mpiexec', '-n', '2', '/usr/bin/python3.9', '/build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/parallel/tests/simple_reduction.py', '--directory=/tmp/tmpag842xi0']
KILLING PROCESS ON TIMEOUT
Rank 0: Generating output in /tmp/tmpag842xi0
Rank 1: Generating output in /tmp/tmpag842xi0

ValueError: Dimension 1 invalid for PyZoltan!
Exception ignored in: 'pyzoltan.core.zoltan.get_geometry_list'
Traceback (most recent call last):
  File "/build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/solver/application.py", line 1353, in _setup_parallel_manager_and_initial_load_balance
    pm.update()
ValueError: Dimension 1 invalid for PyZoltan!

################################################################################
test simple_reduction.py failed with returncode -9
################################################################################
=============================== warnings summary ===============================
../../../../../../../usr/lib/python3/dist-packages/compyle/types.py:164
  /usr/lib/python3/dist-packages/compyle/types.py:164: DeprecationWarning: `np.bool` is a deprecated alias for the builtin `bool`. To silence this warning, use `bool` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.bool_` here.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    np.dtype(np.bool): 'char',

../../../../../../../usr/lib/python3/dist-packages/compyle/types.py:173
  /usr/lib/python3/dist-packages/compyle/types.py:173: DeprecationWarning: `np.bool` is a deprecated alias for the builtin `bool`. To silence this warning, use `bool` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.bool_` here.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    'bool': np.bool,

../../../../../../../usr/lib/python3/dist-packages/pyximport/pyximport.py:51
  /usr/lib/python3/dist-packages/pyximport/pyximport.py:51: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    import imp

../../../../../../../usr/lib/python3/dist-packages/_pytest/mark/structures.py:337
  /usr/lib/python3/dist-packages/_pytest/mark/structures.py:337: PytestCollectionWarning: cannot collect 'test_all_backends' because it is not a function.
    def __call__(self, *args: object, **kwargs: object):

pysph/base/nnps.py:1
  /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/base/nnps.py:1: DeprecationWarning: `np.float` is a deprecated alias for the builtin `float`. To silence this warning, use `float` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.float64` here.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    from pysph.base.nnps_base import get_number_of_threads, py_flatten, \

pysph/examples/tests/test_examples.py:84
  /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/examples/tests/test_examples.py:84: PytestUnknownMarkWarning: Unknown pytest.mark.slow - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @mark.slow

pysph/parallel/tests/test_openmp.py:17
  /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/parallel/tests/test_openmp.py:17: PytestUnknownMarkWarning: Unknown pytest.mark.slow - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @mark.slow

pysph/parallel/tests/test_openmp.py:33
  /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/parallel/tests/test_openmp.py:33: PytestUnknownMarkWarning: Unknown pytest.mark.slow - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @mark.slow

pysph/parallel/tests/test_parallel.py:55
  /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/parallel/tests/test_parallel.py:55: PytestUnknownMarkWarning: Unknown pytest.mark.parallel - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @mark.parallel

pysph/parallel/tests/test_parallel.py:59
  /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/parallel/tests/test_parallel.py:59: PytestUnknownMarkWarning: Unknown pytest.mark.parallel - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @mark.parallel

pysph/parallel/tests/test_parallel.py:72
  /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/parallel/tests/test_parallel.py:72: PytestUnknownMarkWarning: Unknown pytest.mark.slow - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @mark.slow

pysph/parallel/tests/test_parallel.py:73
  /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/parallel/tests/test_parallel.py:73: PytestUnknownMarkWarning: Unknown pytest.mark.parallel - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @mark.parallel

pysph/parallel/tests/test_parallel.py:92
  /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/parallel/tests/test_parallel.py:92: PytestUnknownMarkWarning: Unknown pytest.mark.parallel - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @mark.parallel

pysph/parallel/tests/test_parallel.py:98
  /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/parallel/tests/test_parallel.py:98: PytestUnknownMarkWarning: Unknown pytest.mark.parallel - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @mark.parallel

pysph/parallel/tests/test_parallel.py:112
  /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/parallel/tests/test_parallel.py:112: PytestUnknownMarkWarning: Unknown pytest.mark.parallel - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @mark.parallel

pysph/parallel/tests/test_parallel_run.py:25
  /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/parallel/tests/test_parallel_run.py:25: PytestUnknownMarkWarning: Unknown pytest.mark.slow - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @mark.slow

pysph/parallel/tests/test_parallel_run.py:26
  /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/parallel/tests/test_parallel_run.py:26: PytestUnknownMarkWarning: Unknown pytest.mark.parallel - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @mark.parallel

pysph/parallel/tests/test_parallel_run.py:39
  /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/parallel/tests/test_parallel_run.py:39: PytestUnknownMarkWarning: Unknown pytest.mark.slow - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @mark.slow

pysph/parallel/tests/test_parallel_run.py:40
  /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/parallel/tests/test_parallel_run.py:40: PytestUnknownMarkWarning: Unknown pytest.mark.parallel - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @mark.parallel

pysph/parallel/tests/test_parallel_run.py:50
  /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/parallel/tests/test_parallel_run.py:50: PytestUnknownMarkWarning: Unknown pytest.mark.parallel - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @mark.parallel

<frozen importlib._bootstrap>:228
  <frozen importlib._bootstrap>:228: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.

pysph/tools/tests/test_geometry_stl.py:7
  /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/tools/tests/test_geometry_stl.py:7: DeprecationWarning: `np.float` is a deprecated alias for the builtin `float`. To silence this warning, use `float` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.float64` here.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    import pysph.tools.geometry_stl as G

.pybuild/cpython3_3.9/build/pysph/sph/bc/tests/test_simple_inlet_outlet.py::TestSimpleInlet1D::test_inlet_calls_callback
  /usr/lib/python3/dist-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/.pysph/source/py3.9-linux-x86_64/m_d4d2f4620a7b2524086275ccd7e1980a.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)

.pybuild/cpython3_3.9/build/pysph/sph/bc/tests/test_simple_inlet_outlet.py::TestSimpleOutlet1D::test_outlet_absorbs_particles_from_source
  /usr/lib/python3/dist-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/.pysph/source/py3.9-linux-x86_64/m_e33701e75932aa50c325a9da5905e046.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_acceleration_eval.py::TestAccelerationEval1D::test_should_call_initialize_pair
  /usr/lib/python3/dist-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/.pysph/source/py3.9-linux-x86_64/m_23a3818adde45352a04c1b7364f8beb8.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_acceleration_eval.py::TestAccelerationEval1D::test_should_call_pre_post_functions_in_group
  /usr/lib/python3/dist-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/.pysph/source/py3.9-linux-x86_64/m_6bd24ef6276cfb0a627c4a76f48d9905.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_acceleration_eval.py::TestAccelerationEval1D::test_should_call_py_initialize
  /usr/lib/python3/dist-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/.pysph/source/py3.9-linux-x86_64/m_2050fc7c6320c48dbe6a33538cf05088.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_acceleration_eval.py::TestAccelerationEval1D::test_should_handle_repeated_helper_functions
  /usr/lib/python3/dist-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/.pysph/source/py3.9-linux-x86_64/m_766c702abd060b362383b083d9c03275.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_acceleration_eval.py::TestAccelerationEval1D::test_should_iterate_iterated_group
  /usr/lib/python3/dist-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/.pysph/source/py3.9-linux-x86_64/m_33b81f4dba430aaf42210931044b04c0.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_acceleration_eval.py::TestAccelerationEval1D::test_should_iterate_nested_groups
  /usr/lib/python3/dist-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/.pysph/source/py3.9-linux-x86_64/m_7717c1442a4d826d0e187f434ef96cbc.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_acceleration_eval.py::TestAccelerationEval1D::test_should_not_iterate_normal_group
  /usr/lib/python3/dist-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/.pysph/source/py3.9-linux-x86_64/m_cfe2eb093a39e8e702efa0cf780bf4e3.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_acceleration_eval.py::TestAccelerationEval1D::test_should_run_reduce
  /usr/lib/python3/dist-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/.pysph/source/py3.9-linux-x86_64/m_678795442fd61741ccc5ce5dd5c5c753.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_acceleration_eval.py::TestAccelerationEval1D::test_should_support_constants
  /usr/lib/python3/dist-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/.pysph/source/py3.9-linux-x86_64/m_7bb1bab96b93bbdaafc45e03d4993835.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_acceleration_eval.py::TestAccelerationEval1D::test_should_support_loop_all_and_loop
  /usr/lib/python3/dist-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/.pysph/source/py3.9-linux-x86_64/m_e7e9ab313fa06d6ac295a713d6d20ed0.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_acceleration_eval.py::TestAccelerationEval1D::test_should_support_loop_all_and_loop
  /usr/lib/python3/dist-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/.pysph/source/py3.9-linux-x86_64/m_536b4600578e21ae33f7deb8aa1f71fe.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_acceleration_eval.py::TestAccelerationEval1D::test_should_work_with_non_double_arrays
  /usr/lib/python3/dist-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/.pysph/source/py3.9-linux-x86_64/m_f9827192da77f6672a3575a967f33ba7.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_integrator.py::TestIntegratorAdaptiveTimestep::test_compute_timestep_with_dt_adapt
  /usr/lib/python3/dist-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/.pysph/source/py3.9-linux-x86_64/m_76734bae3a461753975fbf791fdbb19b.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_integrator.py::TestIntegratorAdaptiveTimestep::test_compute_timestep_with_dt_adapt_trumps_dt_cfl
  /usr/lib/python3/dist-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/.pysph/source/py3.9-linux-x86_64/m_3eef2402041a6d8884be144e1ea8a11f.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_integrator.py::TestIntegratorAdaptiveTimestep::test_compute_timestep_with_dt_cfl
  /usr/lib/python3/dist-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/.pysph/source/py3.9-linux-x86_64/m_14fdd08882c92b4eca96b8e5525a5e26.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_integrator.py::TestIntegratorAdaptiveTimestep::test_compute_timestep_without_adaptive
  /usr/lib/python3/dist-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/.pysph/source/py3.9-linux-x86_64/m_a0271db338b0a591d39b669652df021c.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_integrator.py::TestLeapFrogIntegrator::test_helper_can_be_used_with_stepper
  /usr/lib/python3/dist-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/.pysph/source/py3.9-linux-x86_64/m_afce861c4d3d616622e5e739877d460b.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_integrator.py::TestLeapFrogIntegrator::test_integrator_calls_only_py_when_no_stage
  /usr/lib/python3/dist-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/.pysph/source/py3.9-linux-x86_64/m_a8dbf3b89f1f24e1514c5a05249db577.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_integrator.py::TestLeapFrogIntegrator::test_integrator_calls_py_stage1
  /usr/lib/python3/dist-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/.pysph/source/py3.9-linux-x86_64/m_5d94160891ab9d7b65d483394d53f090.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_integrator.py::TestLeapFrogIntegrator::test_integrator_calls_py_stage1_stage2
  /usr/lib/python3/dist-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/.pysph/source/py3.9-linux-x86_64/m_91ff9b09094d9dbbd640f4b6119df413.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_integrator.py::TestLeapFrogIntegrator::test_leapfrog
  /usr/lib/python3/dist-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/.pysph/source/py3.9-linux-x86_64/m_17141a5fd5c62030e087718d7d2cd89f.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_integrator.py::TestPEFRLIntegrator::test_pefrl
  /usr/lib/python3/dist-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/.pysph/source/py3.9-linux-x86_64/m_28bc252d0d0bd9e4d053fe4ec8d0ebc8.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_kernel_corrections.py::TestKernelCorrection2D::test_crksph
  /usr/lib/python3/dist-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/.pysph/source/py3.9-linux-x86_64/m_03fdf2f0a8238e4859e602e5100a8ae8.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_kernel_corrections.py::TestKernelCorrection2D::test_crksph_symmetric
  /usr/lib/python3/dist-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/.pysph/source/py3.9-linux-x86_64/m_3879b817675749e849fa14f03c2f577e.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_kernel_corrections.py::TestKernelCorrection2D::test_gradient_correction
  /usr/lib/python3/dist-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/.pysph/source/py3.9-linux-x86_64/m_c2abdc8087fe603ffd109cd9e5859cec.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_kernel_corrections.py::TestKernelCorrection2D::test_mixed_gradient_correction
  /usr/lib/python3/dist-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/.pysph/source/py3.9-linux-x86_64/m_d286287fc29a88343393d0c2866d769c.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_kernel_corrections.py::TestKernelCorrection3D::test_gradient_correction
  /usr/lib/python3/dist-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/.pysph/source/py3.9-linux-x86_64/m_aa0363728d382da65870dd91f24707af.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_kernel_corrections.py::TestKernelCorrection3D::test_mixed_gradient_correction
  /usr/lib/python3/dist-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/.pysph/source/py3.9-linux-x86_64/m_102aa4de9cd8d5ff37ffec30e2e1c5a5.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_linalg.py::TestLinalg::test_band_matrix
  /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/sph/tests/test_linalg.py:92: PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray.
    mat = np.matrix(mat)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_linalg.py::TestLinalg::test_band_matrix
  /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/sph/tests/test_linalg.py:93: PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray.
    new_b = mat * np.transpose(np.matrix(result))

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_linalg.py::TestLinalg::test_band_matrix
.pybuild/cpython3_3.9/build/pysph/sph/tests/test_linalg.py::TestLinalg::test_dense_matrix
.pybuild/cpython3_3.9/build/pysph/sph/tests/test_linalg.py::TestLinalg::test_general_matrix
.pybuild/cpython3_3.9/build/pysph/sph/tests/test_linalg.py::TestLinalg::test_symmetric_matrix
.pybuild/cpython3_3.9/build/pysph/sph/tests/test_linalg.py::TestLinalg::test_symmetric_positivedefinite_Matrix
.pybuild/cpython3_3.9/build/pysph/sph/tests/test_linalg.py::TestLinalg::test_tridiagonal_matrix
.pybuild/cpython3_3.9/build/pysph/tools/tests/test_geometry.py::TestGeometry::test_rotate
.pybuild/cpython3_3.9/build/pysph/tools/tests/test_geometry.py::TestGeometry::test_rotate
  /usr/lib/python3/dist-packages/numpy/matrixlib/defmatrix.py:69: PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray.
    return matrix(data, dtype=dtype, copy=False)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_linalg.py::TestLinalg::test_dense_matrix
  /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/sph/tests/test_linalg.py:103: PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray.
    mat = np.matrix(mat)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_linalg.py::TestLinalg::test_dense_matrix
  /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/sph/tests/test_linalg.py:104: PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray.
    new_b = mat * np.transpose(np.matrix(result))

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_linalg.py::TestLinalg::test_general_matrix
  /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/sph/tests/test_linalg.py:81: PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray.
    mat = np.matrix(mat)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_linalg.py::TestLinalg::test_general_matrix
  /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/sph/tests/test_linalg.py:82: PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray.
    new_b = mat * np.transpose(np.matrix(result))

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_linalg.py::TestLinalg::test_symmetric_matrix
  /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/sph/tests/test_linalg.py:126: PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray.
    mat = np.matrix(mat)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_linalg.py::TestLinalg::test_symmetric_matrix
  /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/sph/tests/test_linalg.py:127: PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray.
    new_b = mat * np.transpose(np.matrix(result))

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_linalg.py::TestLinalg::test_symmetric_positivedefinite_Matrix
  /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/sph/tests/test_linalg.py:138: PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray.
    mat = np.matrix(mat)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_linalg.py::TestLinalg::test_symmetric_positivedefinite_Matrix
  /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/sph/tests/test_linalg.py:139: PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray.
    new_b = mat * np.transpose(np.matrix(result))

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_linalg.py::TestLinalg::test_tridiagonal_matrix
  /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/sph/tests/test_linalg.py:115: PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray.
    mat = np.matrix(mat)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_linalg.py::TestLinalg::test_tridiagonal_matrix
  /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/sph/tests/test_linalg.py:116: PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray.
    new_b = mat * np.transpose(np.matrix(result))

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_multi_group_integrator.py::TestMultiGroupIntegrator::test_different_accels_per_integrator
  /usr/lib/python3/dist-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/.pysph/source/py3.9-linux-x86_64/m_03d2f3c8840eaa38e2d13edb3a0927a1.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)

.pybuild/cpython3_3.9/build/pysph/sph/tests/test_multi_group_integrator.py::TestMultiGroupIntegrator::test_different_accels_per_integrator
  /usr/lib/python3/dist-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/.pysph/source/py3.9-linux-x86_64/m_f16a981c222b02ffbc9a137fd99ac24b.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)

.pybuild/cpython3_3.9/build/pysph/tools/tests/test_geometry.py::TestGeometry::test_rotate
  /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/tools/geometry.py:176: PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray.
    rotation_matrix = matrix_exp(np.matrix(matrix))

.pybuild/cpython3_3.9/build/pysph/tools/tests/test_geometry.py::TestGeometry::test_rotate
  /build/2/pysph-1.0~b0~20191115.gite3d5e10/2nd/.pybuild/cpython3_3.9/build/pysph/tools/tests/test_geometry.py:76: PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray.
    rotation_matrix = G.matrix_exp(np.matrix(mat))

.pybuild/cpython3_3.9/build/pysph/tools/tests/test_geometry_stl.py::TestGeometry::test_get_points_from_mgrid
  /usr/lib/python3/dist-packages/numpy/core/fromnumeric.py:1970: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.
    result = asarray(a).shape

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED pysph/parallel/tests/test_parallel.py::MPIReduceArrayTestCase::test_parallel_reduce
= 1 failed, 525 passed, 273 skipped, 90 deselected, 77 warnings in 2228.24s (0:37:08) =
```

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -1,3 +1,9 @@
+1.0b2
+-----
+
+* Release date: Still under development.
+
+
 1.0b1
 -----
 

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -1,8 +1,49 @@
 1.0b1
 -----
 
-* Release date: Still under development.
-* Remove pyzoltan, cyarray into their own packages on pypi.
+Around 140 pull requests were merged. Thanks to all who contributed to this
+release (in alphabetical order): Abhinav Muta, Aditya Bhosale, Amal Sebastian,
+Ananyo Sen, Antonio Valentino, Dinesh Adepu, Jeffrey D. Daye, Navaneet, Miloni
+Atal, Pawan Negi, Prabhu Ramachandran, Rohan Kaushik, Tetsuo Koyama, and Yash
+Kothari.
+
+* Release date: 1st March 2022.
+
+* Enhancements:
+
+  * Use github actions for tests and also test OpenCL support on CI.
+  * Parallelize the build step of the octree NNPS on the CPU.
+  * Support for packing initial particle distributions.
+  * Add support for setting load balancing weights for particle arrays.
+  * Use meshio to read data and convert them into particles.
+  * Add support for conditional group of equations.
+  * Add options to control loop limits in a Group.
+  * Add ``pysph binder``, ``pysph cull``, and ``pysph cache``.
+  * Use OpenMP for initialize, loop and post_loop.
+  * Added many SPH schemes: CRKSPH, SISPH, basic ISPH, SWE, TSPH, PSPH.
+  * Added a mirror boundary condition along coordinate axes.
+  * Add support for much improved inlets and outlets.
+  * Add option ``--reorder-freq`` to turn on spatial reordering of particles.
+  * API: Integrators explicitly call update_domain.
+  * Basic CUDA support.
+  * Many important improvements to the pysph Mayavi viewer.
+  * Many improvements to the 3D and 2D jupyter viewer.
+  * ``Application.customize_output`` can be used to customize viewer.
+  * Use ``~/.compyle/config.py`` for user customizations.
+  * Remove pyzoltan, cyarray, and compyle into their own packages on pypi.
+
+* Bug fixes:
+
+  * Fix issue with update_nnps being called too many times when set for a
+    group.
+  * Many OpenCL related fixes and improvements.
+  * Fix bugs in the parallel manager code and add profiling information.
+  * Fix hdf5 compressed output.
+  * Fix ``pysph dump_vtk``
+  * Many fixes to various schemes.
+  * Fix memory leak with the neighbor caching.
+  * Fix issues with using PySPH on FreeBSD.
+
 
 1.0a6
 -----

--- a/pysph/__init__.py
+++ b/pysph/__init__.py
@@ -1,5 +1,5 @@
 # See PEP 440 for more on suitable version numbers.
-__version__ = '1.0b1'
+__version__ = '1.0b2.dev0'
 
 # Utility functions to determine if Zoltan/MPI are available.
 _has_zoltan = None

--- a/pysph/__init__.py
+++ b/pysph/__init__.py
@@ -1,5 +1,5 @@
 # See PEP 440 for more on suitable version numbers.
-__version__ = '1.0b1.dev0'
+__version__ = '1.0b1'
 
 # Utility functions to determine if Zoltan/MPI are available.
 _has_zoltan = None

--- a/pysph/parallel/tests/test_parallel.py
+++ b/pysph/parallel/tests/test_parallel.py
@@ -95,7 +95,8 @@ class MPIReduceArrayTestCase(unittest.TestCase):
     def test_parallel_reduce(self):
         args = ['--directory=%s' % self.root]
         run_parallel_script.run(
-            filename='simple_reduction.py', args=args, nprocs=4, path=path
+            filename='simple_reduction.py', args=args, nprocs=4, path=path,
+            timeout=60.0
         )
 
 

